### PR TITLE
Non bittorrent magnets

### PR DESCRIPTION
**Pull request by KayleePop**
**What is the purpose of this pull request? (put an "X" next to item)**

[ ] Documentation update
[X] Bug fix
[ ] New feature
[ ] Other, please explain:

**What changes did you make? (Give an overview)**

parseTorrent() now throws when passed a non-bittorrent magnet link.

**Which issue (if any) does this pull request address?**

https://github.com/webtorrent/webtorrent/pull/1770

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -21,30 +21,53 @@ module.exports.toTorrentFile = encodeTorrentFile
...
+
     if (!torrentId.urlList) torrentId.urlList = []
+
     return torrentId
-  } else {
-    throw new Error('Invalid torrent identifier')
   }
+
+  // else
+  throw new Error('Invalid torrent identifier')
```
Comment (alxhotel) [inline index.js]: I would rather keep the `else if` structure, I think it was more readable.

Comment (KayleePop) [reply index.js]: I couldn't parse anything because it was so dense D:

should I revert that?

**Comment by KayleePop** [comment]
I wouldn't consider this breaking when the library is "parse torrent" not "parse magnet". Especially when there's a separate module that's specifically for that

**Comment by KayleePop** [comment]
Go ahead and merge when/if you think it's ready :D

File: index.js (hunk 1 above)
Comment (alxhotel) [reply index.js]: It's just a personal preference. You can leave it as it is :)

**Comment by alxhotel** [comment]
> and the readme says "The only property that is guaranteed to be present is infoHash.", so I'd consider returning an object without that property a bug

True, lets merge it :)

File: index.js (hunk 1 above)
Comment (KayleePop) [reply index.js]: Oh, lol I just committed the revert

Comment (alxhotel) [reply index.js]: The timing lol

Any way, thanks for changing it :)

**Comment by feross** [comment]
Just seeing this PR now. Looks good to me! Thanks @alxhotel and @KayleePop!

**Pull request merged**